File: rails/app/models/article_comment.rb
class ArticleComment < ApplicationRecord
  self.table_name = 'article_comments'

  has_many :replies, class_name: 'ArticleComment', inverse_of: :parent_comment
  belongs_to :article, class_name: 'Article', inverse_of: :comments
  belongs_to :author, class_name: 'Player', inverse_of: :article_comments
  belongs_to :parent_comment, class_name: 'ArticleComment', inverse_of: :replies, optional: true

  def to_s
    body.to_s
  end

  # Business operations

  def hide
    # TODO: implement hide
  end

  def unhide
    # TODO: implement unhide
  end

  def is_reply
    # TODO: implement is_reply
    nil
  end

  def as_json(options = {})
    hash = super(options)
    hash['createdAt'] = hash.delete('created_at') if hash.key?('created_at')
    hash
  end
end

File: spring/src/main/java/cardsproject/domain/content/ArticleComment.java
package cardsproject.domain.content;

import jakarta.persistence.*;
import java.time.LocalDateTime;
import com.fasterxml.jackson.annotation.JsonIgnore;
import com.fasterxml.jackson.annotation.JsonProperty;

@Entity
@Table(name = "article_comments")
public class ArticleComment {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String body = "";
    private Boolean isHidden = false;
    private LocalDateTime createdAt;

    // @ManyToOne -> Article, onDelete=CASCADE, relatedName=comments
    @Column(name = "article_id")
    private Long articleId;
    // @ManyToOne -> Player, onDelete=PROTECT, relatedName=article_comments, via=players
    @Column(name = "author_id")
    private Long authorId;
    // @ManyToOne -> ArticleComment, onDelete=SET_NULL, relatedName=replies
    @Column(name = "parent_comment_id")
    private Long parentCommentId;

    public Long getId() { return id; }
    public void setId(Long id) { this.id = id; }
    public String getBody() { return body; }
    public void setBody(String body) { this.body = body; }
    public Boolean getIsHidden() { return isHidden; }
    public void setIsHidden(Boolean isHidden) { this.isHidden = isHidden; }
    @JsonProperty("createdAt")
    public LocalDateTime getCreatedAt() { return createdAt; }
    public void setCreatedAt(LocalDateTime createdAt) { this.createdAt = createdAt; }
    public Long getArticleId() { return articleId; }
    public void setArticleId(Long articleId) { this.articleId = articleId; }
    public Long getAuthorId() { return authorId; }
    public void setAuthorId(Long authorId) { this.authorId = authorId; }
    public Long getParentCommentId() { return parentCommentId; }
    public void setParentCommentId(Long parentCommentId) { this.parentCommentId = parentCommentId; }

    // ── Business operations ──────────────────────────────────────────
    @com.fasterxml.jackson.annotation.JsonIgnore
    public void hide() {
        // TODO: implement hide
    }
    @com.fasterxml.jackson.annotation.JsonIgnore
    public void unhide() {
        // TODO: implement unhide
    }
    @com.fasterxml.jackson.annotation.JsonIgnore
    public Boolean isReply() {
        // TODO: implement isReply
        return null;
    }
}
